src/CareerBoostAI.Application/Candidate/Commands/CreateProfile/CreateProfileCommandHandler.cs

```diff
@@ -0,0 +1,79 @@
+        private readonly ICandidateRepository _candidateRepository;
+        private readonly IEmailSender _emailSender;
+        private readonly ICandidateFactory _candidateFactory;
+        private readonly IUnitOfWork _unitOfWork;
+        public CreateProfileCommandHandler(
+            IFileStorageService fileStorageService, IMediator mediator, 
+            IEmailSender emailSender, ICandidateRepository candidateRepository, 
+            ICandidateReadService candidateReadService,
+            ICandidateFactory candidateFactory, IUnitOfWork unitOfWork)
+        {
+            _emailSender = emailSender;
+            _candidateRepository = candidateRepository;
+            _candidateReadService = candidateReadService;
+            _candidateFactory = candidateFactory;
+            _unitOfWork = unitOfWork;
+        }
+        public async Task Handle(CreateProfileCommand command, CancellationToken cancellationToken)
+        {
+            await Validate(command, cancellationToken);
+            
+            var candidate = CreateAggregateFromCommand(command);
+            
+            await _candidateRepository.CreateNewAsync(candidate);
+            
+            var adminNotificationMessage =
+                $"A new candidate profile has been created for {candidate.FirstName.Value} {candidate.LastName.Value}.";
+            await _emailSender.SendEmailToAdminAsync(subject: "New Candidate Profile Created", body: adminNotificationMessage);
+            
+            
+            await _unitOfWork.SaveChangesAsync(cancellationToken); 
+        }
+
+    private CandidateAggregate CreateAggregateFromCommand(CreateProfileCommand request)
+    {
+        var cv = _candidateFactory.CreateCv(
+            request.Id, request.CvData.Summary,
+            request.CvData.Experiences
+                .Select(exp => ( Guid.NewGuid(),
+                    exp.OrganisationName, exp.City, exp.Country,
+                    exp.StartDate, exp.EndDate, exp.Description,
+                    exp.SequenceIndex)),
+            request.CvData.Educations
+                .Select(edu => (Guid.NewGuid(), edu.OrganisationName, edu.City, edu.Country,
+                    edu.StartDate, edu.EndDate, edu.Program, edu.Grade,
+                    edu.SequenceIndex)),
+            request.CvData.Languages,
+            request.CvData.Skills);
+
+        CandidateAggregate candidate = _candidateFactory
+            .Create(Guid.NewGuid(), request.FirstName, 
+                request.LastName, request.DateOfBirth, 
+                request.Email, request.PhoneCode, request.PhoneNumber, 
+                cv);
+        return candidate;
+    }
+
+    private async Task Validate(CreateProfileCommand request, CancellationToken cancellationToken)
+    {
+        if (await _candidateReadService.CandidateExistsByEmailAsync(request.Email, cancellationToken))
+        {
+            throw new DuplicateCandidateProfileException(request.Email);
```

Comment:
We need to consider if this is expected functionality, the initial idea was for a user to upload a CV and request feedback. Afterward, they may wish to make changes and re-upload their new CV?